server/remote/gitlab/convert.go

```diff
@@ -161,6 +165,7 @@ func convertPushHock(hook *gitlab.PushEvent) (*model.Repo, *model.Build, error)
 	build.Branch = strings.TrimPrefix(hook.Ref, "refs/heads/")
 	build.Ref = hook.Ref
 
+	files := make([]string, 0, len(hook.Commits)*4)
```

Comment:
Why do you used `len(hook.Commits)*4` as capacity?

Reply:
as https://github.com/woodpecker-ci/woodpecker/pull/697#discussion_r786300852

Reply:
```suggestion
        // assume a capacity of 4 changed files per commit
	files := make([]string, 0, len(hook.Commits)*4)
```